commit/merkleroot/processor.go

```diff
@@ -0,0 +1,79 @@
+package merkleroot
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+	"github.com/smartcontractkit/chainlink-ccip/shared"
+)
+
+type Processor struct {
+	nodeID       commontypes.OracleID
+	cfg          pluginconfig.CommitPluginConfig
+	lggr         logger.Logger
+	observer     Observer
+	readerSyncer *plugincommon.BackgroundReaderSyncer
+	ccipReader   reader.CCIP
+	reportingCfg ocr3types.ReportingPluginConfig
+	chainSupport shared.ChainSupport
+}
+
+// NewMerkleWorker creates a new Processor
+func NewMerkleWorker(
```

Comment:
I think just `NewProcessor` is sufficient, no need for the `Merkle`. It'll be called like `merkleroot.NewProcessor(...)`